python/chemiscope/explore/__init__.py

```diff
@@ -1,22 +1,73 @@
+import warnings
+
 from ..jupyter import show
+from ..structures import extract_properties
 from ._metatomic import metatomic_featurizer
 
 
-__all__ = [
-    "explore",
-    "metatomic_featurizer",
-]
+__all__ = ["explore", "metatomic_featurizer", "get_featurizer"]
+
+KNOWN_FEATURIZERS = ["pet-mad-1.0"]
+
+
+def get_featurizer(name):
+    """
+    Get a featurizer by name for feature extraction. Currently available version is:
+    "pet-mad-1.0", which returns an instance of `PETMADFeaturizer
+    <https://arxiv.org/abs/2506.19674>`_.
+
+    :param str name: name of the featurizer. Must match one of the known versions.
+        Currently available is "pet-mad-1.0"
+
+    :param str device: optional. device to run the featurizer on (e.g., "cpu" or "cuda")
+
+    :param int batch_size: optional. number of structures to process per batch during
+        featurization, defaults to 1
+
+    .. warning::
+
+        Requires additional dependencies. Install it using:
+
+        .. code:: bash
+
+            pip install chemiscope[explore]
+    """
+    try:
+        import torch
+        from pet_mad.explore import PETMADFeaturizer
+    except ImportError as e:
+        raise ImportError(
+            "Required package not found. Please install the dependencies with "
+            "`pip install chemiscope[explore]`."
+        ) from e
+
+    if not isinstance(name, str):
+        raise TypeError(f"featurizer name must be a string, not {type(name)}")
+
+    if name == "pet-mad-1.0":
+        device = "cuda" if torch.cuda.is_available() else "cpu"
+        model = PETMADFeaturizer(version="1.0.0", device=device, batch_size=1)
+        return model
+    else:
+        options = (
+            ", ".join(KNOWN_FEATURIZERS)
+            if len(KNOWN_FEATURIZERS) > 1
+            else KNOWN_FEATURIZERS[0]
+        )
+        raise ValueError(
+            f"unknown featurizer: {name}. Available options are: {options}"
+        )
```

Comment:
`join` does nothing with a 1 element list, so you can simplify here:

```suggestion
        raise ValueError(
            f"unknown featurizer: {name}. Available options are: {', '.join(KNOWN_FEATURIZERS)}"
        )
```